src/python_testing/TC_DISHM_2_1.py

```diff
@@ -0,0 +1,281 @@
+            TestStep(3, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"
+                     "Select a value which is NOT in supported_modes_dut and save it as invalid_mode_th"),
+            TestStep(4, "TH sends a ChangeToMode command to the DUT with NewMode set to old_current_mode_dut",
+                     "DUT responds contains a ChangeToModeResponse command with a SUCCESS (value 0x00) status response"),
+            TestStep(5, "Manually put the device in a state from which it will FAIL to transition to PIXIT.DISHM.MODE_CHANGE_FAIL"),
+            TestStep(6, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"),
+            TestStep(7, "TH sends a ChangeToMode command to the DUT with NewMode set to PIXIT.DISHM.MODE_CHANGE_FAIL",
+                     "DUT responds contains a ChangeToModeResponse command with Status field is set to GenericFailure(0x02), InvalidInMode(0x03), or in the MfgCodes (0x80 to 0xBF) range and StatusText field has a length between 1 and 64"),
+            TestStep(8, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and equal to old_current_mode_dut"),
+            TestStep(9, "Manually put the device in a state from which it will SUCCESSFULLY transition to PIXIT.DISHM.MODE_CHANGE_OK"),
+            TestStep(10, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"),
+            TestStep(11, "TH sends a ChangeToMode command to the DUT with NewMode set to PIXIT.DISHM.MODE_CHANGE_OK",
+                     "DUT responds contains a ChangeToModeResponse command with a SUCCESS (value 0x00) status response"),
+            TestStep(12, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and it is the same as step 11"),
+            TestStep(13, "TH sends a ChangeToMode command to the DUT with NewMode set to invalid_mode_th",
+                     "DUT responds contains a ChangeToModeResponse command with a UnsupportedMode(0x01) status response"),
+            TestStep(14, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and same as step 12")
+        ]
+        return steps
+
+    def pics_TC_DISHM_2_1(self) -> list[str]:
+        pics = [
+            "DISHM.S"
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_DISHM_2_1(self):
+
+        class CommonCodes(Enum):
+            SUCCESS = 0x00
+            UNSUPPORTED_MODE = 0x01
+            GENERIC_FAILURE = 0x02
+            INVALID_IN_MODE = 0x03
+
+        cluster = Clusters.Objects.DishwasherMode
+        supported_modes_attribute = cluster.Attributes.SupportedModes
+        current_mode_attribute = cluster.Attributes.CurrentMode
+        endpoint = self.get_endpoint(default=1)
```

Comment:
```suggestion
        endpoint = self.get_endpoint()
```
Goes with the run_if_endpoint... change above since that enforces that an endpoint is set